Lib/test/test_capi/test_float.py

```diff
@@ -0,0 +1,103 @@
+import array
+import math
+import sys
+import unittest
+import warnings
+
+from test.test_capi.test_getargs import (Float, FloatSubclass, FloatSubclass2,
+                                         BadIndex2, BadFloat2, Index, BadIndex,
+                                         BadFloat)
+from test.support import import_helper
+
+_testcapi = import_helper.import_module('_testcapi')
+
+NULL = None
+
+
+class CAPIFloatTest(unittest.TestCase):
+    def assertEqualWithSign(self, actual, expected):
+        self.assertEqual(actual, expected)
+        self.assertEqual(math.copysign(1, actual), math.copysign(1, expected))
+
+    def test_check(self):
+        # Test PyFloat_Check()
+        check = _testcapi.float_check
+
+        self.assertTrue(check(3.14))
+        self.assertTrue(check(FloatSubclass(3.14)))
+        self.assertFalse(check(Float()))
+        self.assertFalse(check(3))
+        self.assertFalse(check([]))
+        self.assertFalse(check(object()))
+
+        # CRASHES check(NULL)
+
+    def test_checkexact(self):
+        # Test PyFloat_CheckExact()
+        checkexact = _testcapi.float_checkexact
+
+        self.assertTrue(checkexact(3.14))
+        self.assertFalse(checkexact(FloatSubclass(3.14)))
+        self.assertFalse(checkexact(Float()))
+        self.assertFalse(checkexact(3))
+        self.assertFalse(checkexact([]))
+        self.assertFalse(checkexact(object()))
+
+        # CRASHES checkexact(NULL)
+
+    def test_fromstring(self):
+        # Test PyFloat_FromString()
+        fromstring = _testcapi.float_fromstring
+
+        self.assertEqual(fromstring("3.14"), 3.14)
+        self.assertEqual(fromstring(b"3.14"), 3.14)
+        self.assertEqual(fromstring(bytearray(b"3.14")), 3.14)
+        self.assertEqual(fromstring(array.array('b', b'3.14')), 3.14)
+        self.assertRaises(TypeError, fromstring, 3.14)
+
+        # CRASHES fromstring(NULL)
+
+    def test_asdouble(self):
+        # Test PyFloat_AsDouble()
+        asdouble = _testcapi.float_asdouble
+
+        class BadFloat3:
+            def __float__(self):
+                raise RuntimeError
+
+        self.assertEqual(asdouble(3.14), 3.14)
+        self.assertEqual(asdouble(-1), -1.0)
```

Comment:
Test also positive integer and -1.0.

Test also with too large integer like 2**1000.